File: collector/src/autostart.test.ts
import { describe, it, expect } from 'vitest';
import { buildScheduledTaskCommand } from './autostart.js';

describe('buildScheduledTaskCommand', () => {
  it('builds a /Create command with ONLOGON trigger, no elevation, quoted paths', () => {
    const argv = buildScheduledTaskCommand('create', 'C:\\Program Files\\nodejs\\node.exe', 'C:\\Users\\test\\aether-os\\collector\\dist\\index.js');
    expect(argv).toEqual([
      '/Create',
      '/TN', 'AetherCollector',
      '/TR', '"C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\test\\aether-os\\collector\\dist\\index.js"',
      '/SC', 'ONLOGON',
      '/RL', 'LIMITED',
      '/F',
    ]);
  });

  it('builds a /Delete command by task name only', () => {
    const argv = buildScheduledTaskCommand('delete', 'unused', 'unused');
    expect(argv).toEqual(['/Delete', '/TN', 'AetherCollector', '/F']);
  });
});

File: collector-go/internal/autostart/autostart_test.go
package autostart

import (
	"reflect"
	"testing"
)

func TestBuildScheduledTaskCommand_Create_OnLogonNoElevationQuotedPaths(t *testing.T) {
	argv := BuildScheduledTaskCommand("create", `C:\Program Files\nodejs\node.exe`, `C:\Users\test\aether-os\collector\dist\index.js`)
	want := []string{
		"/Create",
		"/TN", "AetherCollector",
		"/TR", `"C:\Program Files\nodejs\node.exe" "C:\Users\test\aether-os\collector\dist\index.js"`,
		"/SC", "ONLOGON",
		"/RL", "LIMITED",
		"/F",
	}
	if !reflect.DeepEqual(argv, want) {
		t.Errorf("argv = %#v, want %#v", argv, want)
	}
}

func TestBuildScheduledTaskCommand_Delete_ByTaskNameOnly(t *testing.T) {
	argv := BuildScheduledTaskCommand("delete", "unused", "unused")
	want := []string{"/Delete", "/TN", "AetherCollector", "/F"}
	if !reflect.DeepEqual(argv, want) {
		t.Errorf("argv = %#v, want %#v", argv, want)
	}
}
